dot/rpc/modules/grandpa.go

```diff
@@ -65,7 +65,7 @@ func (gm *GrandpaModule) ProveFinality(r *http.Request, req *ProveFinalityReques
 	}
 
 	// Leaving check in for linter
-	if req.authorityID != uint64(0) { // nolint:staticcheck
+	if req.authorityID != uint64(0) { //// nolint:staticcheck
```

Comment:
I would like to suggest to leave these `//nolint` comments here as they were before, and comment out the linters not working with Go 1.18 in `.golangci-lint.yml`, and add custom rules to `nolintlint` such that it ignores `//nolint` comments for these disabled linters.

I'll monitor when these get fixed to work with generics and will re-add them over time.